Cmake does not work any more with Visual C++ 2022

Using Visual C++ 2022, I have the following error message
      [CMake] -- The CXX compiler identification is unknown
      [CMake] CMake Error at D:\Users\Public\Documents\DevGit\khiops\CMakeLists.txt:10 (project):
      [CMake]   The CMAKE_CXX_COMPILER:
      [CMake] 
      [CMake]     cl
      [CMake] 
      [CMake]   is not a full path and was not found in the PATH.
      [CMake] 
      [CMake]   Tell CMake where to find the compiler by setting either the environment
      [CMake]   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path

It worked before on my PC. 
It is broken since I had a patch of Windows and  the following update of Visual C++ 2022:
  Microsoft Visual Studio Professional 2022
  Version 17.6.5
  VisualStudio.17.Release/17.6.5+33829.357
  Microsoft .NET Framework
  Version 4.8.04084
  Version installée : Professional
  ...

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -49,7 +49,8 @@
             "Windows"
           ]
         }
-      }
+      },
+      "generator": "Visual Studio 17 2022"
     },
     {
       "name": "macos",

